src/main/java/lotto/validator/NumberValidator.java

```diff
@@ -0,0 +1,45 @@
+package lotto.validator;
+
+import java.util.List;
+import lotto.exception.ErrorMessage;
+import lotto.exception.InvalidInputException;
+
+public class NumberValidator {
+    private final int value;
+
+    private NumberValidator(int value) {
+        this.value = value;
+    }
+
+    public static NumberValidator of(int value) {
+        return new NumberValidator(value);
+    }
+
+    public NumberValidator shouldPositive(ErrorMessage errorMessage) {
+        if (value < 0) {
+            throw new InvalidInputException(errorMessage);
+        }
+        return this;
+    }
```

Comment:
```suggestion
    public NumberValidator shouldPositive() {
        if (value < 0) {
            throw new InvalidInputException(ErrorMessage.INPUT_NOT_POSITIVE_NUMBER);
        }
        return this;
    }
```

이렇게 하지 않고 매개변수로 받아서 출력하신 이유가 뭔가요??

Reply:
validator에서 다양한 에러 메세지가 나올 수 있다고 생각했습니다.
지금은 "입력값은 자연수여야 합니다"로 받고 있지만 "로또 번호는 자연수여야 합니다"가 될 수도 있고, "보너스 번호는 자연수여야 합니다."로 변할 수도 있다고 생각했습니다.